HW0/P4/P4.py

```diff
@@ -29,49 +29,118 @@
     #
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
+
+read_temp=np.loadtxt('P4_trajectory.txt',delimiter=',')
+xx=(np.transpose(read_temp))[0]
+yy=(np.transpose(read_temp))[1]
+zz=(np.transpose(read_temp))[2]
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+ax.plot(xx, yy, zz,'--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
-    #####################
+read_temp_meas=np.loadtxt('P4_measurements.txt',delimiter=',')
+C=np.matrix([[1/rx,0,0],[0,1/ry,0],[0,0,1/rz]])
+sm=np.transpose(read_temp_meas)
+xk=C*sm
+x_apx=np.squeeze(np.asarray(xk[0]))
+y_apx=np.squeeze(np.asarray(xk[1]))
+z_apx=np.squeeze(np.asarray(xk[2])) 
+	#####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+ax.plot(x_apx, y_apx, z_apx,'.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
-
-    # A = ?
-    # a = ?
+s0=np.matrix([[0],[0],[2],[15],[3.5],[4.0]])
+A=np.matrix([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt,0,0],[0,0,0,0,1-c*dt,0],[0,0,0,0,0,1-c*dt]])
+a=np.matrix([[0],[0],[0],[0],[0],[g*dt]])
+s=np.zeros((120,6))
```

Comment:
There is an easier way of creating A and a matrix that might help you in the future:
A = np.matrix(np.zeros([6,6]))
np.fill_diagonal(A, [1,1,1,1 - cdt,1 - cdt,1 - c*dt])

A[0,3] = dt
A[1,4] = dt
A[2,5] = dt

 a = np.matrix(np.zeros([6,1]))
a[5,0] = g*dt